### ISSUE TITLE
Unexpected failure due to HO with Bags.

### ISSUE THREAD

**Issue by aniemetz** [cvc5/cvc5-projects#394]
cvc5/cvc5@7f4fe916ceda4a42ccbad740d620e5ef852f0280
murxla/murxla@998f16907d49fc0e3b81f122d5c89ae33d2e1ee8
```
TEST_F(TestApiBlackSolver, foo)
{
  Solver slv;
  Sort s1 = slv.getBooleanSort();
  Sort s2 = slv.mkFunctionSort({s1}, s1);
  Sort s3 = slv.getRealSort();
  Sort s4 = slv.mkBagSort(s3);
  Sort s5 = slv.mkArraySort(s4, s4);
  Sort s6 = slv.mkBagSort(s1);
  Sort s7 = slv.mkSetSort(s6);
  Sort s9 = slv.mkSequenceSort(s5);
  Term t1 = slv.mkConst(s6, "_x8");
  Term t3 = slv.mkConst(s2, "_x10");
  Term t6 = slv.mkTerm(Kind::BAG_MAP, {t3, t1});
  Term t7 = slv.mkTerm(Kind::BAG_DUPLICATE_REMOVAL, {t1});
  Term t8 = slv.mkConst(s7, "_x38");
  Term t9 = slv.mkTerm(Kind::SET_MEMBER, {t6, t8});
  slv.checkEntailed({t9});
}
```
Unexpectedly fails with
```
Partial function applications are only supported with higher-order logic. Try adding the logic prefix HO_.
```

**Comment by ajreynol** [comment]
After discussing with @mudathirmahgoub , this is actually the correct behavior.  `t3` is being used as a partial application in the argument to Kind::BAG_MAP.

**Comment by aniemetz** [comment]
I don't understand why this is a partial application. Can you explain?

**Comment by ajreynol** [comment]
It is partial since it is not applied to any arguments.

**Comment by aniemetz** [comment]
Murxla configuration fixed in murxla/murxla@664478e77d4735dc1a6b5aeb53caf3eb16065e6b.
